quickwit/quickwit-codegen-example/src/hello.rs

```diff
@@ -61,6 +68,60 @@ impl tower::Service<HelloRequest> for HelloClient {
         Box::pin(fut)
     }
 }
+#[derive(Debug)]
+struct MailboxAdapter<A: quickwit_actors::Actor, E> {
+    inner: quickwit_actors::Mailbox<A>,
+    phantom: std::marker::PhantomData<E>,
+}
+impl<A, E> std::ops::Deref for MailboxAdapter<A, E>
+where
+    A: quickwit_actors::Actor,
+{
+    type Target = quickwit_actors::Mailbox<A>;
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+#[derive(Debug)]
+pub struct HelloMailbox<A: quickwit_actors::Actor> {
+    inner: MailboxAdapter<A, crate::HelloError>,
+}
+impl<A: quickwit_actors::Actor> HelloMailbox<A> {
+    pub fn new(instance: quickwit_actors::Mailbox<A>) -> Self {
+        let inner = MailboxAdapter {
+            inner: instance,
+            phantom: std::marker::PhantomData,
+        };
+        Self { inner }
+    }
+}
+impl<A, M, T, E> tower::Service<M> for HelloMailbox<A>
+where
+    A: quickwit_actors::Actor + quickwit_actors::Handler<M, Reply = Result<T, E>> + Send
+        + Sync + 'static,
+    M: std::fmt::Debug + Send + Sync + 'static,
+    T: Send + Sync + 'static,
+    E: std::fmt::Debug + Send + Sync + 'static,
+    crate::HelloError: From<quickwit_actors::AskError<E>>,
+{
+    type Response = T;
+    type Error = crate::HelloError;
+    type Future = BoxFuture<Self::Response, Self::Error>;
+    fn poll_ready(
+        &mut self,
+        _cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Result<(), Self::Error>> {
+        //! This no longer works with parallel actors. The fix is to acquire a permit from the mailbox in `poll_ready` and consumer it in `call`.
```

Comment:
I don't understand the "parallel actors" reference. Are you referring to a specific implementation (possibly on an old branch)?

Reply:
I updated the comment. I was trying to say that this approach does not work with "balance" middleware such as `tower::balance::pool::Pool`.